### PR TITLE
mvcc/reader: ignore lock of lock type for read.

### PR DESCRIPTION
**Pull request by coocood**
Same effect as https://github.com/pingcap/tikv/pull/3011

### CODE REVIEW

File: src/storage/mvcc/reader.rs [hunk 1]
```diff
@@ -194,34 +194,32 @@ impl MvccReader {
         Ok(Some((commit_ts, write)))
     }
 
-    fn check_lock(&mut self, key: &Key, mut ts: u64) -> Result<Option<u64>> {
+    fn check_lock(&mut self, key: &Key, mut ts: u64) -> Result<bool> {
```
Comment (BusyJay) [inline src/storage/mvcc/reader.rs]: What does `bool` mean?

File: src/storage/mvcc/reader.rs [hunk 2]
```diff
@@ -194,34 +194,33 @@ impl MvccReader {
...
+        Ok(())
     }
 
     pub fn get(&mut self, key: &Key, mut ts: u64) -> Result<Option<Value>> {
         // Check for locks that signal concurrent writes.
         match self.isolation_level {
-            IsolationLevel::SI => if let Some(new_ts) = self.check_lock(key, ts)? {
-                ts = new_ts;
-            },
+            IsolationLevel::SI => self.check_lock(key, ts)?,
```
Comment (AndreMouche) [inline src/storage/mvcc/reader.rs]: What if there are lots of `Rollback ` during  (lock.ts ~ ts)? The related issue is https://github.com/pingcap/tikv/pull/2871

File: src/storage/mvcc/reader.rs [hunk 3]
```diff
@@ -194,34 +194,33 @@ impl MvccReader {
...
-                    return Err(Error::KeyIsLocked {
-                        key: key.raw()?,
-                        primary: lock.primary,
-                        ts: lock.ts,
-                        ttl: lock.ttl,
-                    });
-                }
+            if lock.ts > ts ||
+                lock.lock_type == LockType::Lock ||
+                ts == u64::MAX && key.raw()? == lock.primary {
```
Comment (AndreMouche) [inline src/storage/mvcc/reader.rs]: Do one thing in one PR? I do not think is a good idea to change (ts==u64::MAX)'s logic in this PR.

Comment (coocood) [reply src/storage/mvcc/reader.rs]: Just make the logic consistent.

File: src/storage/mvcc/reader.rs (hunk 2 above)
Comment (coocood) [reply src/storage/mvcc/reader.rs]: Then we just scan and ignore those rollbacks.
In the normal case, there is no lock, we need to scan those rollbacks anyway.
I think this optimization doesn't worth the complexity cost.

**Pull request closed without merging**